src/app/space/forge-wizard/abstract-wizard.component.ts

```diff
@@ -87,8 +88,12 @@ export abstract class AbstractWizard implements OnInit {
     });
     obs.subscribe(
       codebase => {
-        // todo broadcast
-        // this._broadcaster.broadcast('codebaseAdded', codebase);
+        if (codebase.attributes.type === 'git') {
+          codebase.name = codebase.attributes.url.replace('.git', '').replace('git@github.com:', '');
+        } else {
+          codebase.name = codebase.attributes.url;
+        }
+        this.broadcaster.broadcast('codebaseAdded', codebase);
```

Comment:
Although this works for the existing/old forge wizard, we also need a solution that works with the new app-launcher. 

In addition to broadcasting here, we also need to listen for the onComplete event generated by the new app-launcher component. Then, broadcast the same event from the create-app and import-app components of app-launcher.

Reply:
Okay, I see where this logic would be added, but I'm unsure why it's required. Since the new app-launcher is not just an overlay over the Analyze page but actually routes the user away to the applauncher and back again upon completion, will the widget ever actually receive this broadcast from the app launcher? It appears that the workflow would include the widget being destroyed and then recreated after the user has completed (or exited) the wizard, in which case the onInit logic should already restore the correct state. Is this incorrect?

And if that's how the new app launcher works, then perhaps we can just drop this patch, since the bug is only possible with the old wizard?

Reply:
If the widget is being recreated, then it should update properly without a broadcast. Please disregard my comments